### PR TITLE
Unitary: Add generalized Hadamard gate

### PR DESCRIPTION
**Pull request by madcpf**
- Add generalized Hadamard gate into qudit_gates and qudit_effects
- Fix a few typos in qudit_gates and qudit_effects

### CODE REVIEW

File: unitary/alpha/qudit_effects.py [hunk 1]
```diff
@@ -127,3 +125,27 @@ class QuditFlip(Flip):
...
+
+    def effect(self, *objects):
+        for q in objects:
+            if q.qubit.dimension == 2:
+                yield cirq.H(q.qubit)
+            else:
+                yield QuditHadamardGate(dimension=q.qubit.dimension)(q.qubit)
+
+
+class QuditSuperpose(Superpose):
```
Comment (dstrain115) [inline unitary/alpha/qudit_effects.py:143]: For what backwards compatibility?  This is a new gate.  Let's remove this.

File: unitary/alpha/qudit_effects.py [hunk 2]
```diff
@@ -127,3 +125,27 @@ class QuditFlip(Flip):
 
     def __init__(self, dimension: int, state0: int, state1: int):
         super().__init__(state0=state0, state1=state1)
+
+
+class Superpose(QuantumEffect):
```
Comment (dstrain115) [inline unitary/alpha/qudit_effects.py:130]: Let's move this into unitary.alpha.qubit_effects.Superposition if possible.

The feedback from test users is that no one wants to worry about Qudits vs Qubits, and that we should be able to do Superpostion() and it should work on whatever dimension (which the qudit_gate you wrote already supports)

File: unitary/alpha/qudit_gates.py [hunk 3]
```diff
@@ -34,28 +34,32 @@ def __init__(
...
         return (self.dimension,)
 
     def _unitary_(self):
-        arr = np.zeros((self.dimension, self.dimension))
-        arr[self.source_state, self.destination_state] = 1
-        arr[self.destination_state, self.source_state] = 1
-        for i in range(self.dimension):
-            if i != self.source_state and i != self.destination_state:
-                arr[i, i] = 1
+        arr = np.eye(self.dimension)
```
Comment (dstrain115) [inline unitary/alpha/qudit_gates.py:46]: These are great changes, but let's split them out into a different PR since they are not relevant to the creation of a generalized Hadamard.

File: unitary/alpha/qudit_effects.py (hunk 1 above)
Comment (madcpf) [reply unitary/alpha/qudit_effects.py]: Thanks Doug. It's removed.

File: unitary/alpha/qudit_effects.py (hunk 2 above)
Comment (madcpf) [reply unitary/alpha/qudit_effects.py]: Thanks Doug. It's combined into qubit_effects.Superposition.

File: unitary/alpha/qudit_gates.py (hunk 3 above)
Comment (madcpf) [reply unitary/alpha/qudit_gates.py]: Thanks Doug. They are split out and will be shared in another PR.

**Review by dstrain115 — APPROVED** [review]
